packages/kit/src/core/dev/plugin.js

```diff
@@ -165,10 +166,12 @@ export async function create_plugin(config, cwd) {
 							? await vite.ssrLoadModule(`/${config.kit.files.hooks}`)
 							: {};
 
+						const handle = user_hooks.handle || (({ event, resolve }) => resolve(event));
+
 						/** @type {import('types/internal').Hooks} */
 						const hooks = {
 							getSession: user_hooks.getSession || (() => ({})),
-							handle: user_hooks.handle || (({ event, resolve }) => resolve(event)),
+							handle: amp ? sequence(handle, amp) : handle,
```

Comment:
i think these need to be reversed

```suggestion
							handle: amp ? sequence(amp, handle) : handle,
```

Reply:
I believe they're applied left-to-right. Wouldn't we want the amp validator to run last? I.e. after the HTML is completely generated make sure it's valid

Reply:
they're not applied left-to-right _or_ right-to-left. it just applies the first in the sequence, which is responsible for calling the second (which is responsible for calling the third, etc). so `amp` needs to call the user's `handle`, and inspect the result of calling that

Reply:
done